src/modules/Form/index.tsx

```diff
@@ -0,0 +1,199 @@
+import { CommonFieldType, CustomPokemon, GenderFieldType, SelectFieldType } from 'types';
+import React, { createRef, FormEvent, RefObject } from 'react';
+import { Layout } from 'modules';
+import * as S from './styled';
+import { uuid } from 'utils';
+import {
+  BirthdayField,
+  ConsentField,
+  GenderField,
+  AvatarField,
+  ShinyField,
+  TypeField,
+  NameField,
+  FormCard,
+} from './components';
+
+type Props = {
+  componentName: string;
+  location: string;
+};
+
+type State = {
+  isShinyField: Pick<CommonFieldType, 'inputRef'>;
+  pokemons: Array<CustomPokemon & { id: string }>;
+  successMessage: RefObject<HTMLSpanElement>;
+  birthdayField: CommonFieldType;
+  consentField: CommonFieldType;
+  genderField: GenderFieldType;
+  avatarField: CommonFieldType;
+  typeField: SelectFieldType;
+  nameField: CommonFieldType;
+  initialEnter: boolean;
+  totalErrors: number;
+};
+
+export class Form extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      pokemons: [],
+      totalErrors: 0,
+      initialEnter: true,
+      successMessage: createRef(),
+      birthdayField: {
+        inputRef: createRef(),
```

Comment:
https://ru.reactjs.org/docs/refs-and-the-dom.html#creating-refs 
you can check that refs and state is different things =) 
And we are not creating refs in state =)
Sometimes ref can hold state value, but not state - ref value)